src/modules/job-manager/alloc.c

```diff
@@ -98,16 +98,20 @@ static void interface_teardown (struct alloc *alloc, char *s, int errnum)
 
 /* Send sched.free request.
  */
-int free_request (struct alloc *alloc, flux_jobid_t id, json_t *R)
+int free_request (struct alloc *alloc,
+                  flux_jobid_t id,
+                  json_t *R,
+                  bool final)
 {
     flux_msg_t *msg;
 
     if (!(msg = flux_request_encode ("sched.free", NULL)))
         return -1;
     if (flux_msg_pack (msg,
-                       "{s:I s:O}",
+                       "{s:I s:O s:b}",
                        "id", id,
-                       "R", R) < 0)
+                       "R", R,
+                       "final", final ? 1 : 0) < 0)
```

Comment:
I've seen this pattern a few places, this is redundant due to int promotion in variadics right or is there a janssonism here I don't know about?

Reply:
You're right - I wasn't using my brain at some point and then after that, monkey see, monkey do.  It should be fine due to integer promotion.  The varidaic unpack is where you could get into trouble since jansson booleans unpack to integers not bools. Will fix.

Reply:
Oh yeah, we actually had a nasty stack smash bug in sched from exactly that issue on the unpack side.  Makes sense.

Reply:
Yeah, I think we've hit _very_ difficult to track down bugs in the unpack side and are a little too wary of pack/unpack of booleans with jansson.